packages/react-native-gesture-handler/apple/RNGestureHandlerDetector.mm

```diff
@@ -0,0 +1,147 @@
+
+    for (const auto handler : props.handlerTags) {
+      NSNumber *handlerTag = [NSNumber numberWithInt:handler];
+      [handlerManager.registry detachHandlerWithTag:handlerTag];
+    }
+  }
+}
+
+- (void)dispatchStateChangeEvent:(RNGestureHandlerDetectorEventEmitter::OnGestureHandlerStateChange)event
+{
+  if (_eventEmitter != nullptr) {
+    std::dynamic_pointer_cast<const RNGestureHandlerDetectorEventEmitter>(_eventEmitter)
+        ->onGestureHandlerStateChange(event);
+  }
+}
+
+- (void)dispatchGestureEvent:(RNGestureHandlerDetectorEventEmitter::OnGestureHandlerEvent)event
+{
+  if (_eventEmitter != nullptr) {
+    std::dynamic_pointer_cast<const RNGestureHandlerDetectorEventEmitter>(_eventEmitter)->onGestureHandlerEvent(event);
+  }
+}
+
+- (void)dispatchTouchEvent:(RNGestureHandlerDetectorEventEmitter::OnGestureHandlerTouchEvent)event
+{
+  if (_eventEmitter != nullptr) {
+    std::dynamic_pointer_cast<const RNGestureHandlerDetectorEventEmitter>(_eventEmitter)
+        ->onGestureHandlerTouchEvent(event);
+  }
+}
+
+#pragma mark - RCTComponentViewProtocol
+
++ (ComponentDescriptorProvider)componentDescriptorProvider
+{
+  return concreteComponentDescriptorProvider<RNGestureHandlerDetectorComponentDescriptor>();
+}
+
+- (void)updateLayoutMetrics:(const LayoutMetrics &)layoutMetrics
+           oldLayoutMetrics:(const LayoutMetrics &)oldLayoutMetrics
+{
+  auto newLayoutMetrics = layoutMetrics;
+  // Override to force hittesting to work outside bounds
+  newLayoutMetrics.overflowInset = {.left = 1, .right = 1, .top = 1, .bottom = 1};
+
+  [super updateLayoutMetrics:newLayoutMetrics oldLayoutMetrics:oldLayoutMetrics];
+}
+
+- (void)updateProps:(const Props::Shared &)propsBase oldProps:(const Props::Shared &)oldPropsBase
+{
+  const auto &newProps = *std::static_pointer_cast<const RNGestureHandlerDetectorProps>(propsBase);
+  const auto &oldProps = *std::static_pointer_cast<const RNGestureHandlerDetectorProps>(oldPropsBase);
+
+  _moduleId = newProps.moduleId;
+  RNGestureHandlerManager *handlerManager = [RNGestureHandlerModule handlerManagerForModuleId:_moduleId];
+  react_native_assert(handlerManager != nullptr && "Tried to access a non-existent handler manager")
+
+      std::unordered_map<int, RNGestureHandlerMutation>
+          changes;
+
+  if (oldPropsBase != nullptr) {
+    for (const auto oldHandler : oldProps.handlerTags) {
+      changes[oldHandler] = RNGestureHandlerMutationDetach;
+    }
+  }
+
+  for (const auto newHandler : newProps.handlerTags) {
+    changes[newHandler] = changes.contains(newHandler) ? RNGestureHandlerMutationKeep : RNGestureHandlerMutationAttach;
+  }
+
+  for (const auto handlerChange : changes) {
+    NSNumber *handlerTag = [NSNumber numberWithInt:handlerChange.first];
+
+    if (handlerChange.second == RNGestureHandlerMutationAttach) {
+      // TODO: Attach to the child when NativeGestureHandler remounts, track children changes?
```

Comment:
😄

Reply:
Updated the todo in [077f1dc](https://github.com/software-mansion/react-native-gesture-handler/pull/3599/commits/077f1dc2049f9feb1c681194dfe7884be8601557). I think that's best done outside of this PR. Getting native gestures to work with the new detector may be tricky as we will need to attach to the child directly instead of the detector itself.